provider/offline.go

```diff
@@ -551,28 +560,34 @@ func (schema *TableSchema) ToParquetRecords(records []GenericRecord) []any {
 			// To ensure the struct fields are public and accessible to other methods,
 			// we need to title case them when setting them.
 			colName := caser.String(schema.Columns[j].Name)
+			parquetField := parquetRecord.Elem().FieldByName(colName)
+			var reflectValue reflect.Value
 			switch v := value.(type) {
-			case int:
-				parquetRecord.Elem().FieldByName(colName).Set(reflect.ValueOf(&v))
-			case int32:
-				parquetRecord.Elem().FieldByName(colName).Set(reflect.ValueOf(&v))
-			case int64:
-				parquetRecord.Elem().FieldByName(colName).Set(reflect.ValueOf(&v))
-			case float32:
-				parquetRecord.Elem().FieldByName(colName).Set(reflect.ValueOf(&v))
-			case float64:
-				parquetRecord.Elem().FieldByName(colName).Set(reflect.ValueOf(&v))
-			case string:
-				parquetRecord.Elem().FieldByName(colName).Set(reflect.ValueOf(&v))
-			case bool:
-				parquetRecord.Elem().FieldByName(colName).Set(reflect.ValueOf(&v))
+			case int, int32, int64, float32, float64, string, bool:
+				// Pointer types are used for all the scalar types to ensure they
+				// can be nullable in the parquet file.
+				ogVal := reflect.ValueOf(v)
+				if ogVal.CanAddr() {
+					reflectValue = ogVal.Addr()
+				} else {
+					reflectValue = reflect.New(reflect.TypeOf(v))
```

Comment:
i wonder if it makes sense to ever cache this? Don't know how inefficient it is in go but when doing someting similar like this in python for numpy it ended up being a bit of a bottlneck